### PR TITLE
Add flake8-docstring to the mix (and declare we're using google style docstrings)

### PR DESCRIPTION
**Pull request by thejcannon**
NOTE: This PR is against `thejcannon/pyflakes`. When the associated PR is merged in I will change to be `ni/master`

This PR adds `flake-docstrings` to the `flake8` plugins we use for enforcement.

Changes to the conventions doc:
- Explicitly called out following google-style docstrings
- Reordered docstring conventions to roughly follow general conventions followed by more technical conventions

Note the following checks are still "on" in the tool but not called out in the convention as they're kind of "syntactic" checks around the docstrings:

D402    First line should not be the function’s “signature”
D403    First word of the first line should be properly capitalized
D405    Section name should be properly capitalized
D410    Missing blank line after section
D411    Missing blank line before section
D412    No blank lines allowed between a section header and its content
D414    Section has no content
D416    Section name should end with a colon
D417    Missing argument descriptions in the docstring
D418    Function/ Method decorated with @overload shouldn’t contain a docstring

### CODE REVIEW

**Comment by thejcannon** [comment]
Abandoning. No way to re-base the PR on a different repo and don't want to make people duplicate their responses.

File: docs/Coding-Conventions.md [hunk 1]
```diff
@@ -853,117 +853,130 @@ def go_shopping():
 
 ## [D.1] Docstrings
 
-### [D.1.1] ✔️ **DO** Write docstrings for all public packages, modules, functions, classes, and methods
+### [D.1.1] ✔️ **DO** Follow Google's docstring conventions
+
+Google's docstring conventions are described here: https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings
```
Comment (mshafer-NI) [inline docs/Coding-Conventions.md:858]: Did we give reST any consideration? (https://stackoverflow.com/a/24385103)

Comment (thejcannon) [reply docs/Coding-Conventions.md:858]: Hmmmmm, yeah I flip flop. Perhaps I'll punt again until later :)

Comment (DavidCurtiss) [reply docs/Coding-Conventions.md:858]: I like the readability of Google's. Why do you prefer reST's style? Conciseness? More widespread?

Comment (mshafer-NI) [reply docs/Coding-Conventions.md:858]: conciseness - and with type-hinting, Google's is overkill.

Comment (DavidCurtiss) [reply docs/Coding-Conventions.md:858]: Google's format doesn't require you to provide the types. And with sphinx_autodoc_typehints, the generated docs still include the type-hinted types. Ex: [this](https://python-docs.systemlink.io/en/stable/api_reference/tag.html#systemlink.clients.tag.TagValueReader.read), generated from [this](https://python-docs.systemlink.io/en/stable/_modules/systemlink/clients/tag/_tag_value_reader.html#TagValueReader.read).

**Pull request closed without merging**